browser/ui/views/playlist/playlist_action_bubble_view.cc

```diff
@@ -437,19 +496,22 @@ PlaylistActionBubbleView* PlaylistActionBubbleView::GetBubble() {
 
 PlaylistActionBubbleView::PlaylistActionBubbleView(
     Browser* browser,
-    views::View* anchor,
+    PlaylistActionIconView* anchor,
     playlist::PlaylistTabHelper* playlist_tab_helper)
     : BubbleDialogDelegateView(anchor, views::BubbleBorder::Arrow::TOP_RIGHT),
       browser_(browser),
-      playlist_tab_helper_(playlist_tab_helper) {}
+      playlist_tab_helper_(playlist_tab_helper),
+      icon_view_(anchor) {}
 
 PlaylistActionBubbleView::~PlaylistActionBubbleView() = default;
 
 void PlaylistActionBubbleView::WindowClosing() {
   BubbleDialogDelegateView::WindowClosing();
 
-  DCHECK_EQ(g_bubble, this);
-  g_bubble = nullptr;
+  // g_bubble could be different one when new bubble is showing up subsequently.
+  if (g_bubble == this) {
```

Comment:
I'm confusing between here and https://github.com/brave/brave-core/pull/19290/files#diff-1172219cf20634416f0b31a88f83d802a3bfded69ef8fa30cabf96cda8de08fcR41

In ShowBubble(), it checks `g_bubble` is null by `DCHECK(!g_bubble);`
It means, we assume that there is no active bubble when showing new bubble.
But, I think this check(`g_bubble == this`) means another bubble could be already shown
before closing previous bubble.
To me both checking seems contradict. confusing.. :)

Reply:
Yeah, it's quite confusing. As we're calling OnWindowClosing() twice, `g_bubble != this` is true on the second call.

>  Call sequence | g_bubble state
> 
>  ---
> 
>  Close callback
>     WindowClosing | |this| to null
>     ShowBubble    | |null| to new bubble
>  WindowClosing callback
>     WindowClosing | shouldn't be changed.

For clarity, extract some part of code into a separate method - WindowClosingImpl().